### ISSUE TITLE
Running the setup doesn't create a new repository

### ISSUE THREAD

**Issue by SVAR7BERG** [react-boilerplate/react-boilerplate#2192]
## Issue Type

- [ ] Feature (https://github.com/react-boilerplate/react-boilerplate/blob/master/.github/CONTRIBUTING.md#feature-requests)
- [X] Bug (https://github.com/react-boilerplate/react-boilerplate/blob/master/.github/CONTRIBUTING.md#bug-reports)

## Description

`npm run setup` on windows doesn't create a new repository after installing the dependencies and clearing the old repository.

## Steps to reproduce

```js
git clone --depth=1 https://github.com/react-boilerplate/react-boilerplate.git

npm run setup
- or -
yarn setup

(when asked if you want to clear the old repo, say "Y")
```

It installs the dependencies and says "initializing new repository" but doesn't actually create one. The will be no .git folder afterwards.

# Versions

- React-Boilerplate (see `package.json`): 3.5.0
- Node/NPM: 8.9.1 / 5.7.1
- Browser: /
- OS: Windows
- CLI: Git Bash / MINGW / Windows CMS / Windows Powershell (tested even more)

**Comment by julienben** [comment]
Which version of Windows? Also, did you see if the command `git init && git add . && git commit -m "Initial commit"` appeared in your terminal? If it did then most likely the problem is elsewhere. Can you try running `git init` from a different folder?

**Comment by SVAR7BERG** [comment]
Tested it on Win7 and Win10.

The command itself doesn't appear in the terminal.

> Can you try running git init from a different folder?
I can even do `git init` in the root folder of the project. The old repo _does_ get removed. Thats not the problem. But no new one is being created.

Here's my whole console log of setting up react-boilerplate:

```
VDL2@VDL2 MINGW64 /d/daten-VDL2/Projects
$ git clone --depth=1 https://github.com/react-boilerplate/react-boilerplate.git
Cloning into 'react-boilerplate'...
remote: Counting objects: 379, done.
remote: Compressing objects: 100% (316/316), done.
Receiving objects: 100% (379/379), 1.16 MiB | 1.63 MiB/s, done.d 0R

Resolving deltas: 100% (63/63), done.

VDL2@VDL2 MINGW64 /d/daten-VDL2/Projects
$ cd react-boilerplate/

VDL2@VDL2 MINGW64 /d/daten-VDL2/Projects/react-boilerplate (master)
$ yarn setup
yarn setup v1.0.2
$ npm i chalk shelljs
npm notice created a lockfile as package-lock.json. You should commit this file.
+ chalk@1.1.3
+ shelljs@0.7.7
added 23 packages from 8 contributors in 3.416s
$ node ./internals/scripts/setup.js


Do you want to clear old repository? [Y/n] Y
Removing old repository ✓
Installing dependencies ✓. (This might take a while)

Initialising new repository
$ npm run build:dll

> react-boilerplate@3.5.0 build:dll D:\daten-VDL2\Projects\react-boilerplate
> node ./internals/scripts/dependencies.js

Building the Webpack DLL...
Hash: 0a2a831b54673a41eff7
Version: webpack 3.5.5
Time: 5622ms
                      Asset  Size  Chunks                    Chunk Names
reactBoilerplateDeps.dll.js  3 MB       0  [emitted]  [big]  reactBoilerplateDeps
chunk    {0} reactBoilerplateDeps.dll.js (reactBoilerplateDeps) 2.49 MB [entry] [rendered]
Done in 181.32s.

VDL2@VDL2 MINGW64 /d/daten-VDL2/Projects/react-boilerplate
$ ls -la
total 895
drwxr-xr-x 1 VDL2 197121      0 Apr 24 14:25 .
drwxr-xr-x 1 VDL2 197121      0 Apr 24 14:24 ..
-rw-r--r-- 1 VDL2 197121    222 Apr 24 14:25 .editorconfig
-rw-r--r-- 1 VDL2 197121   1577 Apr 24 14:25 .gitattributes
drwxr-xr-x 1 VDL2 197121      0 Apr 24 14:25 .github
-rw-r--r-- 1 VDL2 197121    132 Apr 24 14:25 .gitignore
-rw-r--r-- 1 VDL2 197121    430 Apr 24 14:25 .travis.yml
drwxr-xr-x 1 VDL2 197121      0 Apr 24 14:25 app
-rw-r--r-- 1 VDL2 197121   1296 Apr 24 14:25 appveyor.yml
-rw-r--r-- 1 VDL2 197121  10445 Apr 24 14:25 Changelog.md
-rw-r--r-- 1 VDL2 197121   2440 Apr 24 14:25 CODE_OF_CONDUCT.md
-rw-r--r-- 1 VDL2 197121   6138 Apr 24 14:25 CONTRIBUTING.md
drwxr-xr-x 1 VDL2 197121      0 Apr 24 14:25 docs
drwxr-xr-x 1 VDL2 197121      0 Apr 24 14:25 internals
-rw-r--r-- 1 VDL2 197121   1106 Apr 24 14:25 LICENSE.md
drwxr-xr-x 1 VDL2 197121      0 Apr 24 14:27 node_modules
-rw-r--r-- 1 VDL2 197121   8829 Apr 24 14:25 package.json
-rw-r--r-- 1 VDL2 197121   6071 Apr 24 14:25 package-lock.json
-rw-r--r-- 1 VDL2 197121  10639 Apr 24 14:25 README.md
drwxr-xr-x 1 VDL2 197121      0 Apr 24 14:25 server
-rw-r--r-- 1 VDL2 197121 294321 Apr 24 14:27 yarn.lock

VDL2@VDL2 MINGW64 /d/daten-VDL2/Projects/react-boilerplate
$
```

**Comment by julienben** [comment]
After you do `cd react-boilerplate`, you do `yarn setup`, `npm i chalk shelljs` and then `node ./internals/scripts/setup.js`. I'm not sure precisely how this differs from the recommended steps but I'm not surprised it has a different result. Can you try from scratch using the steps listed [here](https://github.com/react-boilerplate/react-boilerplate#quick-start)? Also, don't do `yarn setup`. `npm run setup` will use `yarn` automatically if it finds it.

**Comment by SVAR7BERG** [comment]
> npm i chalk shelljs and then node ./internals/scripts/setup.js
are actually performed by yarn setup itself. I already tried it with `npm run setup` before. So I actually did what the quick start says. It just doesn't create a new git as is inteded (i guess, from looking at the script setup.js itself)

**Comment by SVAR7BERG** [comment]
Btw, after running `npm run setup` which - as mentioned - fails to create a new git repo, I tried to run `npm run clean` which gave me this:

```
Cleanup started... ✓fatal: Not a git repository (or any of the parent directories): .git
```

**Comment by julienben** [comment]
I don't have a Windows env to reproduce this so you're more than welcome to debug this on your own and report issues and/or potential solutions.

If I were you I'd start by running the following script in a folder without git and take it from there:

```js
const exec = require('child_process').exec;

function initGit() {
  exec('git init && git add . && git commit -m "Initial commit"');
}

initGit()
```

Or you could put console.logs in `setup.js` until you find the point where the script stops behaving as expected.

**Comment by SVAR7BERG** [comment]
Yeah I might try that. Although, if I choose `no` when asked to clean the old repo and run `npm run clean` afterwards, it seems to work fine.

Do you know whats the actual difference?

**Comment by julienben** [comment]
The `clean` script doesn't do anything to your project's git setup (except a single commit).

You should also test [this line](https://github.com/react-boilerplate/react-boilerplate/blob/master/internals/scripts/setup.js#L46). If it doesn't work on your side (probably permissions-related in that case), then `git init` won't work after that.

**Comment by Mensae** [comment]
Funny, I never knew we have this feature :-), as it never worked for me either, Windows 10.

I did some debugging but didn't find the problem, yet, but I can confirm that calling the initGit function alone does work.:

```
exec('git init && git add . && git commit -m "Initial commit"', (err, stdout, stderr) => {
  if (err) {
    process.stderr.write(err);
    process.exit(1);
  }
}
```
or wrapping it in a try-catch didn't reveal any problems.

So for the moment I'm in the dark too.

**Comment by gretzky** [comment]
Great find. I don't have a windows env to test in either, but anyone who does that can take a look deeper is more than welcome.

**Comment by mciastek** [comment]
I think that the problem here can be lack of `process.exit` in `initGit` callback (https://github.com/react-boilerplate/react-boilerplate/blob/master/internals/scripts/setup.js#L120-L122). It's possible that whole process will finish before git commands are done.

Here's a fix:

```js
  deleteFileInCurrentDir('setup.js', function () {
    if(clearRepo) {
      interval = animateProgress('Initialising new repository');
      process.stdout.write('Initialising new repository');
      initGit(function () {
        clearInterval(interval);
        process.stdout.write('\nDone!');
        process.exit(0);
      });
    } else {
      process.stdout.write('\nDone!');
      process.exit(0);
    }
  });
```

**Comment by julienben** [comment]
Thanks for the suggestion @mciastek !

@Mensae or @SVAR7BERG Could either you check this on your Windows setup and confirm that this fix works for you too?

**Comment by Mensae** [comment]
@julienben Yes, I can confirm it works!

Thanks @mciastek I planned to have look into it but did'nt find the time. Can you create a PR?

**Comment by julienben** [comment]
Any PRs in the next few days should target the v3.6 branch so it can easily be included in the upcoming release.

**Comment by mciastek** [comment]
Great! @julienben I made a PR with fix - #2245.

**Comment by SVAR7BERG** [comment]
I'm glad you could already confirm it as I didn't find time to do so, yet. Thanks you very much!

**Comment by julienben** [comment]
I merged #2245 into the v3.6 branch so it will be released soon. Closing this. Thanks everyone.
